enos/modules/install_packages/scripts/distro-repo-setup.sh

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env bash
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: BUSL-1.1
+
+set -e
+
+fail() {
+  echo "$1" 1>&2
+  exit 1
+}
+
+[[ -z "$DISTRO" ]] && fail "DISTRO env variable has not been set"
+[[ -z "$RETRY_INTERVAL" ]] && fail "RETRY_INTERVAL env variable has not been set"
+[[ -z "$TIMEOUT_SECONDS" ]] && fail "TIMEOUT_SECONDS env variable has not been set"
+
+setup_repos() {
+  # If we don't have any repos on the list for this distro, no action needed.
+  if [ ${#DISTRO_REPOS[@]} -lt 1 ]; then
+    echo "DISTRO_REPOS is empty; No repos required for the packages for this Linux distro."
+    return 0
+  fi
+
+  # Wait for cloud-init to finish so it doesn't race with any of our package installations.
+  # Note: Amazon Linux 2 throws Python 2.7 errors when running `cloud-init status` as
+  # non-root user (known bug).
+  sudo cloud-init status --wait
+
+  if [ "${DISTRO}" == "sles" ]; then
```

Comment:
Not totally sure I understand your comment but here's some further explanation: I structured this script by the distro, because currently `sles` is the only one that requires repo setup. I wouldn't want to define the case by the package manager, as `leap` also uses zypper but does not require repo setup. My idea was that if other future distros require similar repo setup, we could add a case in the script for that distro. Does that address your concern?

Reply:
For posterity: we discussed this and came to an understanding :D I have updated the script to handle the cases of other distros.